designer/client/src/PageCreate.tsx

```diff
@@ -131,6 +136,8 @@ export class PageCreate extends Component<Props, State> {
       // Remove default controller
       if (payload.controller === ControllerType.Page) {
         delete pageNew.controller
+      } else if (payload.controller === ControllerType.Repeat) {
+        ;(pageNew as PageRepeat).repeat.options.name = payload.repeatName
```

Comment:
Can we review the page types here or narrow down `pageNew` to infer it as `PageRepeat`?

Then we don't need to cast